### ISSUE TITLE
dotnet nuget push with wildcards and snupkgs does not push all packages

### ISSUE THREAD

**Issue by Happypig375** [NuGet/Home#9867]
## Details about Problem

NuGet product used (NuGet.exe | VS UI | Package Manager Console | dotnet.exe): dotnet.exe

NuGet version (x.x.x.xxx): 5.7.0.4

dotnet.exe --version (if appropriate): 5.0.100-preview.6.20318.15

VS version (if appropriate): N/A

OS version (i.e. win10 v1607 (14393.321)): win10 v1909 (18363.959)

Worked before? If so, with which NuGet version: N/A

## Detailed repro steps so we can see the same problem

![image](https://user-images.githubusercontent.com/19922066/89126986-a444e300-d51c-11ea-8324-8d367a6ed652.png)

```
hadri@Home MINGW64 ~/source/repos/CSharpMath (master)
$ dotnet nuget push .nupkgs/*.nupkg -k [REDACTED] -s https://api.nuget.org/v3/index.json
Pushing CSharpMath.0.5.0-alpha2.2.nupkg to 'https://www.nuget.org/api/v2/package'...
  PUT https://www.nuget.org/api/v2/package/
  Created https://www.nuget.org/api/v2/package/ 1666ms
Your package was pushed.
Pushing CSharpMath.0.5.0-alpha2.2.snupkg to 'https://www.nuget.org/api/v2/symbolpackage'...
  PUT https://www.nuget.org/api/v2/symbolpackage/
  Created https://www.nuget.org/api/v2/symbolpackage/ 425ms
Your package was pushed.

hadri@Home MINGW64 ~/source/repos/CSharpMath (master)
$ dotnet nuget push .nupkgs/CSharpMath.*.nupkg -k [REDACTED] -s https://api.nuget.org/v3/index.json
Pushing CSharpMath.0.5.0-alpha2.2.nupkg to 'https://www.nuget.org/api/v2/package'...
  PUT https://www.nuget.org/api/v2/package/
  Conflict https://www.nuget.org/api/v2/package/ 1497ms
To skip already published packages, use the option --skip-duplicate
error: Response status code does not indicate success: 409 (A package with ID 'CSharpMath' and version '0.5.0-alpha2.2' already exists and cannot be modified.).


Usage: dotnet nuget push [arguments] [options]

Arguments:
  [root]  Specify the path to the package and your API key to push the package to the server.

Options:
  -h|--help                      Show help information
  --force-english-output         Forces the application to run using an invariant, English-based culture.
  -s|--source <source>           Package source (URL, UNC/folder path or package source name) to use. Defaults to DefaultPushSource if specified in NuGet.Config.
  -ss|--symbol-source <source>   Symbol server URL to use.
  -t|--timeout <timeout>         Timeout for pushing to a server in seconds. Defaults to 300 seconds (5 minutes).
  -k|--api-key <apiKey>          The API key for the server.
  -sk|--symbol-api-key <apiKey>  The API key for the symbol server.
  -d|--disable-buffering         Disable buffering when pushing to an HTTP(S) server to decrease memory usage.
  -n|--no-symbols                If a symbols package exists, it will not be pushed to a symbols server.
  --no-service-endpoint          Does not append "api/v2/package" to the source URL.
  --interactive                  Allow the command to block and require manual action for operations like authentication.
  --skip-duplicate               If a package and version already exists, skip it and continue with the next package in the push, if any.

hadri@Home MINGW64 ~/source/repos/CSharpMath (master)
$ dotnet nuget push .nupkgs/CSharpMath.E*.nupkg -k [REDACTED] -s https://api.nuget.org/v3/index.json
Pushing CSharpMath.Editor.0.5.0-alpha2.2.nupkg to 'https://www.nuget.org/api/v2/package'...
  PUT https://www.nuget.org/api/v2/package/
  Created https://www.nuget.org/api/v2/package/ 1633ms
Your package was pushed.
Pushing CSharpMath.Editor.0.5.0-alpha2.2.snupkg to 'https://www.nuget.org/api/v2/symbolpackage'...
  PUT https://www.nuget.org/api/v2/symbolpackage/
  Created https://www.nuget.org/api/v2/symbolpackage/ 424ms
```

In the first command input, I tried to upload all the packages in the folder. However, it only uploaded the first one found. The second command confirms that the first package did successfully upload. The third command confirms that only the first package uploaded.

## Other suggested things

Workaround is to do command trickery to list files then upload packages one by one instead of the entire folder.

### Verbose Logs

`-d` enables diagnostic output, however no additional info was provided:

```
hadri@Home MINGW64 ~/source/repos/CSharpMath (master)
$ dotnet -d nuget push .nupkgs/CSharpMath.*a*alpha*.nupkg -k [REDACTED] -s https://api.nuget.org/v3/index.json
Telemetry is: Enabled
Pushing CSharpMath.Avalonia.0.5.0-alpha2.2.nupkg to 'https://www.nuget.org/api/v2/package'...
  PUT https://www.nuget.org/api/v2/package/
  Created https://www.nuget.org/api/v2/package/ 1573ms
Your package was pushed.
Pushing CSharpMath.Avalonia.0.5.0-alpha2.2.snupkg to 'https://www.nuget.org/api/v2/symbolpackage'...
  PUT https://www.nuget.org/api/v2/symbolpackage/
  Created https://www.nuget.org/api/v2/symbolpackage/ 454ms
Your package was pushed.
```

### Sample Project

https://github.com/verybadcat/CSharpMath/tree/337d49363e45aaa8fa3414f5d0c0ca497cf594e9

**Comment by Happypig375** [comment]
<s>What's interesting is that using wildcards works for GitHub Packages, but not NuGet.org:
https://github.com/verybadcat/CSharpMath/blob/FixNuGetForReal/.github/workflows/Test.yml#L77-L78</s>

Edit:
When changed the path from `.nupkgs/` to `.nupkgs/*.nupkg` to avoid GitHub Packages interpreting `.snupkg`s as `.nupkg`s, the same bug noted in this issue happened.

**Comment by aortiz-msft** [comment]
Seems duplicate of https://github.com/NuGet/Home/issues/4393

**Comment by dbsanfte** [comment]
I think this issue was opened because someone closed #4393 . That issue has been reopened so no need for a dupe here.
